### PR TITLE
Fix over flow

### LINKED ISSUE #2410: Crash when YIELD -1 * 1
**Your Environments (__must be provided__)**

docker

**How To Reproduce(__must be provided__)**

image

**Expected behavior**

 return -1

### PR DESCRIPTION
**Pull request by laura-ding**
closed #2410

### CODE REVIEW

File: src/common/filter/Expressions.cpp [hunk 1]
```diff
@@ -867,7 +867,7 @@ OptVariantType ArithmeticExpression::eval(Getters &getters) const {
         } else if (lv > 0 && rv < 0) {
             return minInt / lv > rv;
         } else if (lv < 0 && rv > 0) {
-            return minInt / lv < rv;
+            return minInt / rv > lv;
```
Comment (panda-sheep) [inline src/common/filter/Expressions.cpp:870]: What's the difference from the original

File: src/common/filter/Expressions.cpp [hunk 2]
```diff
@@ -929,10 +929,16 @@ OptVariantType ArithmeticExpression::eval(Getters &getters) const {
...
 
-                if (abs(asInt(r)) == 0) {
+                int64_t lValue = asInt(l);
+                int64_t rValue = asInt(r);
+                if (abs(rValue) == 0) {
                     // When Null is supported, should be return NULL
                     return Status::Error("Division by zero");
                 }
+
+                if (lValue == minInt && rValue == -1) {
```
Comment (panda-sheep) [inline src/common/filter/Expressions.cpp:939]: line 929 need to handle.

File: src/common/filter/Expressions.cpp (hunk 1 above)
Comment (laura-ding) [reply src/common/filter/Expressions.cpp:870]: The minInt div a negative int(-1) is overflow.

File: src/common/filter/Expressions.cpp (hunk 2 above)
Comment (laura-ding) [reply src/common/filter/Expressions.cpp:939]: why? double is no need

**Review by panda-sheep — APPROVED** [review]
Good

**Review by bright-starry-sky — APPROVED** [review]
LGTM

**Pull request merged**